### ISSUE TITLE
Nan in Dice

### ISSUE THREAD

**Issue by deepsea920415** [MIC-DKFZ/nnUNet#2132]
![Screenshot_1](https://github.com/MIC-DKFZ/nnUNet/assets/155721694/e899bec4-03be-4ca1-8694-1524be08b559)
This is my training screen.
Why is this happened?
I was separating individual teeth.
There are 33 classes(0: background, 1~32: teeth) but in major images have less teeth than 32.
Is this reason?

Hope to get answer.
I want to use this wonderful neural network in my dental automatic project.

**Comment by constantinulrich** [comment]
Hi, no worries. 
During validation after each epoch nnunet calculates the pseudo dice for 50*batchsize random patches. If the class was not sampled you receive a nan. But that does not impact your training.
Best 
Constantin

**Comment by deepsea920415** [comment]
Thank you very much for your replay.
And I have another question regarding the labeling.

My data label is like as follow:
"background": 0,
"upper 1": 1,
"upper 2": 2,
"upper 3": 3,
"upper 4": 4,
"upper 5": 5,
"upper 6": 6,
"upper 7": 7,
"upper 8": 8,
"upper 9": 9,
"upper 10": 10,
"upper 11": 11,
"upper 12": 12,
"upper 13": 13,
"upper 14": 14,
"upper 15": 15,
"upper 16": 16,
"lower 1": 17,
"lower 2": 18,
"lower 3": 19,
"lower 4": 20,
"lower 5": 21,
"lower 6": 22,
"lower 7": 23,
"lower 8": 24,
"lower 9": 25,
"lower 10": 26,
"lower 11": 27,
"lower 12": 28,
"lower 13": 29,
"lower 14": 30,
"lower 15": 31,
"lower 16": 32

here, order is from right to left.
But in normally people have about 28~30 teeth.
So, some class is missing in major images.
Anyway, is it okay to use nnunet in this case of segmentation task?

Thank you.

**Comment by constantinulrich** [comment]
It is completely okay if one patient does not have all classes. However, from our experience multiclass teeth segmentation is a very difficult task. There have been and there are this year again some miccai challenges about that you might check out to find inspiration what to do to get better results results

**Comment by deepsea920415** [comment]
Thank you very much.
